lib/control/control-server.c

```diff
@@ -138,7 +173,8 @@ static void
 _delete_thread_command_runner(gpointer data)
 {
   ThreadedCommandRunner *self = (ThreadedCommandRunner *) data;
-  g_thread_join(self->thread);
+  g_assert(self->real_thread.tid_saved == TRUE);
+  thread_cancel(self->real_thread.tid);
```

Comment:
A `g_thread_join` is still needed.

Reply:
That would be a bug - join is a blocking call and intentionally replaced by `thread_cancel`.